### PR TITLE
Parser

### PR DESCRIPTION
**Pull request by s4ayub**

### CODE REVIEW

**Comment by jwang61** [comment]
This closes #4

File: src/Parse.hs [hunk 1]
```diff
@@ -0,0 +1,99 @@
+module Parse (parse)  where
+
+import Control.Monad.State (State, get, put, evalState)
+
+import qualified Token as Tok
+import qualified AST as AST
```
Comment (jwang61) [inline src/Parse.hs]: Don't need `as AST`

File: src/Lex.hs [hunk 2]
```diff
@@ -1,19 +1,19 @@
-module Lex where
+module Lex (Lex.lex) where
 
 import qualified Data.Text as T
 import qualified Data.Char as Char
```
Comment (jwang61) [inline src/Lex.hs:4]: Don't need `as Char`

**Review by jwang61 — CHANGES_REQUESTED** [review]
image

Please see requested changes.

File: src/Lex.hs (hunk 2 above)
Comment (jwang61) [reply src/Lex.hs:4]: Take that back, you do.

Comment (s4ayub) [reply src/Lex.hs:4]: you're crazy

**Review by jwang61 — APPROVED** [review]

**Pull request merged**